www/api/class/centreon_monitoring_externalcmd.class.php

```diff
@@ -61,34 +61,78 @@ public function __construct()
      */
     public function postSend()
     {
-        if (isset($this->arguments['commands']) && is_array($this->arguments['commands'])) {
+        global $centreon;
+
+        if (
+            isset($this->arguments['commands'])
+            && is_array($this->arguments['commands'])
+            && count($this->arguments['commands'])
+        ) {
             /* Get poller Listing */
             $query = 'SELECT id ' .
                 'FROM nagios_server ' .
                 'WHERE ns_activate = "1"';
 
             $dbResult = $this->pearDB->query($query);
             $pollers = array();
+
             while ($row = $dbResult->fetchRow()) {
                 $pollers[$row['id']] = 1;
             }
 
-            if (count($this->arguments['commands'])) {
-                if ($fh = @fopen($this->centcore_file, 'a+')) {
-                    foreach ($this->arguments['commands'] as $command) {
-                        if (isset($pollers[$command['poller_id']])) {
-                            fwrite($fh,
-                                "EXTERNALCMD:" . $command["poller_id"] . ":[" .
-                                $command['timestamp'] . "] " . $command['command'] . "\n"
+            $externalCommand = new CentreonExternalCommand();
+            $availableCommands = array();
+
+            /**
+             * We need to make the concordance between the data saved in the database
+             * and the action provided by the user.
+             */
+            foreach ($externalCommand->getExternalCommandList() as $key => $cmd) {
+                foreach ($cmd as $c) {
+                    $availableCommands[$c] = $key;
+                }
+            }
+
+            $isAdmin = $centreon->user->admin;
+
+            /**
+             * If user is not admin we need to retrieve its ACL
+             */
+            if (!$isAdmin) {
+                $userAcl = new CentreonACL($centreon->user->user_id, $centreon->centreon->admin);
+            }
+
+            if ($fh = @fopen($this->centcore_file, 'a+')) {
+                foreach ($this->arguments['commands'] as $command) {
+                    $commandSplitted = explode(';', $command['command']);
+                    $action = $commandSplitted[0];
+
+                    // checking that action provided exists
+                    if (!array_key_exists($action, $availableCommands)) {
+                        throw new RestBadRequestException('Action ' . $action . ' not supported');
+                    }
+
+                    if (!$centreon->user->admin) {
```

Comment:
why declare $isAdmin if you don't use it ?
https://github.com/centreon/centreon/pull/9366/files#diff-5b28191bbed53758319095816ed02e7311c3120918388e11c7f1b0e86899af9aR96